examples/closure_compiled/BUILD.bazel

```diff
@@ -0,0 +1,40 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_closure//closure:defs.bzl", "closure_js_binary")
+load("//:defs.bzl", "ts_library")
+
+ts_library(
+    name = "child",
+    srcs = [
+        "greeter.ts",
+        "declarations.d.ts",
+    ],
+)
+
+ts_library(
+    name = "parent",
+    srcs = ["main.ts"],
+    deps = [":child"],
+)
+
+closure_js_binary(
+    name = "closure_compiled",
+    deps = [":parent"],
+    defs = [
+        "--jscomp_off=reportUnknownTypes",
```

Comment:
I think we should preemptively change this to --jscomp_off=checkTypes . I think that flag disables all the flags that are going to bite us in the future as we expand this example more.

Reply:
Do we need that? tsickle should suppress checkTypes on each JS file:
https://github.com/angular/tsickle/blob/8ebfd011e9386997ca94001210596c5fd75cd17d/test_files/default/default.js#L3

How do we deal with `defs` in general? I think internally at Google we've been learning towards wrapping with a macro like `ts_binary` to make things more uniform. But in open-source there is no uniformity.

/cc @evmar @rkirov

Reply:
We may need both. The generated output definitely has suppression on checkTypes, but without this it is failing to compile ATM